### ISSUE TITLE
overfitting? or not

### ISSUE THREAD

**Issue by CanoeFZH** [hexiangnan/attentional_factorization_machine#5]
python AFM.py --dataset ml-tag --epoch 20 --pretrain 0 --batch_size 4096 --hidden_factor '[16,16]' --keep '[1.0,0.5]' --lamda_attention 100.0 --lr 0.1
 #params: 1537870
```
Init: 	 train=1.0000, validation=1.0000 [3.2 s]
Epoch 1 [4.3 s]	train=0.5210, validation=0.5746 [4.4 s]
Epoch 2 [4.4 s]	train=0.4683, validation=0.5514 [4.3 s]
Epoch 3 [4.8 s]	train=0.4326, validation=0.5401 [4.5 s]
Epoch 4 [5.0 s]	train=0.4055, validation=0.5321 [4.7 s]
Epoch 5 [5.4 s]	train=0.3874, validation=0.5274 [5.2 s]
Epoch 6 [5.0 s]	train=0.3712, validation=0.5232 [5.3 s]
Epoch 7 [5.0 s]	train=0.3558, validation=0.5204 [4.9 s]
Epoch 8 [5.3 s]	train=0.3460, validation=0.5181 [4.9 s]
Epoch 9 [5.4 s]	train=0.3375, validation=0.5163 [5.1 s]
Epoch 10 [5.4 s]	train=0.3287, validation=0.5151 [5.0 s]
Epoch 11 [5.4 s]	train=0.3242, validation=0.5136 [5.0 s]
Epoch 12 [5.1 s]	train=0.3168, validation=0.5126 [5.5 s]
Epoch 13 [5.1 s]	train=0.3119, validation=0.5118 [5.1 s]
Epoch 14 [5.5 s]	train=0.3074, validation=0.5113 [5.1 s]
Epoch 15 [5.5 s]	train=0.3068, validation=0.5106 [5.2 s]
Epoch 16 [5.5 s]	train=0.3022, validation=0.5103 [5.3 s]
Epoch 17 [5.6 s]	train=0.2984, validation=0.5098 [5.3 s]
Epoch 18 [5.2 s]	train=0.2964, validation=0.5092 [5.7 s]
Epoch 19 [5.3 s]	train=0.2934, validation=0.5093 [5.3 s]
Epoch 20 [5.7 s]	train=0.2908, validation=0.5089 [5.3 s]
```

**Comment by hexiangnan** [comment]
As you can see the convergence has slowed down, it is overfitting under this set of params. Try these:
#1: enlarge the hidden_factor(e.g. [32,256]) as it improve the capability of the model;
#2: train AFM based on the pretrained parameters from FM, which makes great difference;
#3: run more epochs(e.g. we set epoch to 100 in our experiments, and unless there is an obvious overfitting the trainning process will last 100 epoches before it stops).

**Comment by CanoeFZH** [comment]
OK, but the training loss and validation loss gap is still huge I think.

**Comment by hexiangnan** [comment]
Yes.
